proxy/proxy.go

```diff
@@ -172,9 +172,10 @@ func (p *Proxy) Handle(w http.ResponseWriter, r *http.Request) {
 			w.WriteHeader(http.StatusAccepted)
 			w.Write([]byte(""))
 			return
-		} else { //If Jenkins is up, we can simply proxy through
-			log.Info(fmt.Sprintf("Passing through %s", r.URL.String()))
 		}
+		//If Jenkins is up, we can simply proxy through
```

Comment:
I was hoping we could also split here - https://github.com/fabric8-services/fabric8-jenkins-proxy/pull/109/files#diff-3ac192716a1df66bbd1008d327af8792R112, almost splitting the method in half. Something like this:

```golang
   if isGH {
       handleGitHubRequest(w, r)
   } else {
       handleDirectJenkinsUIRequest(w,r)
   }
```

This way the existing comments on this if/else branch become obsolete, since the code gives you the same information.

Reply:
Similar to the above comment - https://github.com/fabric8-services/fabric8-jenkins-proxy/pull/109/files#diff-3ac192716a1df66bbd1008d327af8792R153

Here we could extract everything into `if isIdle` into a method `sendStatusAcceptedToClient`.

Reply:
I wonder whether the _redirect_ flag would not be better named as _needsAuth_ or if you want to keep the redirect part _needsAuthRedirect_.

Reply:
What's about extracting a processCookies and processJSONToken as well?

Reply:
Thanks! Will look into it